### PR TITLE
Feature/linaria

### PR DESCRIPTION
**Pull request by nicholasio**
This PR adds support for linaria, which is commonly used to share styled react components between front-end and backend (Gutenberg).

### Description of the Change

Instructs 10up-toolkit on how to compile Linaria code.

### Possible Drawbacks

Had to use my fork of the react-fast-refresh-webpack plugin until https://github.com/pmmmwh/react-refresh-webpack-plugin/pull/726 gets merged.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.
- [x] I have added a changeset to my PR. See [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document for instructions

### CODE REVIEW

**Comment by nicholasio** [comment]
This is blocked by https://github.com/pmmmwh/react-refresh-webpack-plugin/pull/726

File: packages/toolkit/config/webpack/modules.js
Comment (nicholasio) [inline packages/toolkit/config/webpack/modules.js:112]: thread-loader is incompatible with linaria (apparently there's a way but don't feel like it's worth it). I added a `noop-loader` just in case projects are manipulating `rules` and if simply remove we might introduce off-by-one errors.

I never really understood the benefits of thread-loader so I'm not concerned about us just ditching it.... I believe this is a relatively safe change that shouldn't be breaking but open to your thoughts @Antonio-Laguna

**Comment by nicholasio** [comment]
not sure why tests are failing

UPDATE: fixed.

File: packages/toolkit/README.md
Comment (Antonio-Laguna) [inline packages/toolkit/README.md]: [nit]

```suggestion
10up-toolkit supports [Linaria](https://linaria.dev/) without any additional configuration. The main usecase for supporting linaria is to easily compile React components authored using Linaria. This can be useful if you're building a headless site and want to share React components between your front-end and Gutenberg.
```

File: packages/toolkit/README.md
Comment (Antonio-Laguna) [inline packages/toolkit/README.md]: [nit]

```suggestion
If you want 10up-toolkit to compile Linaria first install the following packages:
```

File: packages/toolkit/README.md
Comment (Antonio-Laguna) [inline packages/toolkit/README.md]: [nit]

```suggestion

Once those packages are installed, 10up-toolkit will now look for any Linaria usage in JavaScript files and compile the CSS. The compiled CSS will have the name of the JS entry point (but with a .css extension).
```

File: packages/toolkit/README.md
Comment (Antonio-Laguna) [inline packages/toolkit/README.md]: [nit]

```suggestion
10up-toolkit will compile any Linaria code from the `index.js` entry point to `index.css`. Note that CSS imported by JS entry points always gets generated with the name of the entry point e.g: `index.js => index.css`. So make sure `editorStyle` points to the right CSS file which should be based on `editorScript`.
```

File: packages/toolkit/README.md
Comment (Antonio-Laguna) [inline packages/toolkit/README.md]:
```suggestion
**IMPORTANT**: We do not currently recommend using Linaria for standard WordPress builds. In a headless build the block rendering and front-end styles are handled outside of WordPress, hence why the `block.json` above only cares about the editor script and style.
```

File: packages/toolkit/config/webpack/modules.js
Comment (Antonio-Laguna) [reply packages/toolkit/config/webpack/modules.js:112]: I think you're correct! I don't believe this is breaking! The only effect is that it _might_ (only might) reduce speed by a tiny bit given there are no extra workers.

File: packages/toolkit/package.json
Comment (Antonio-Laguna) [inline packages/toolkit/package.json:23]: Is this fork related to this change? I'm mainly curious

**Review by Antonio-Laguna — DISMISSED** [review]
This LGTM!

I've added some minor nitpick formatting with casing and such. I did double-check that on their repo's README they always capitalize. Not a blocker by any means!

File: packages/toolkit/package.json
Comment (nicholasio) [reply packages/toolkit/package.json:23]: Yes - hopefully we can go back to upstream once this is merged and released https://github.com/pmmmwh/react-refresh-webpack-plugin/pull/726

**Pull request merged**
